internal/obsreportconfig/obsmetrics/obsmetrics.go

```diff
@@ -7,6 +7,7 @@
 package obsmetrics // import "go.opentelemetry.io/collector/internal/obsreportconfig/obsmetrics"
 
 const (
-	NameSep = "/"
-	Scope   = "go.opentelemetry.io/collector/obsreport"
+	SpanNameSep   = "/"
+	MetricNameSep = "."
```

Comment:
This won't be a breaking change for end users who scrape the metrics with prometheus right? But it will be a breaking change for anyone using the otel go sdk to emit metrics right? I am worried about breaking end-user telemetry in a single release, should we add a feature gate?

Reply:
Isn't OTel Go SDK support under a feature gate already?

Reply:
+1 to tyler's concern here. By making the separator here a `.` instead of a `_` we would then create three possibilities for users today:
```
otelcol.process_uptime # the metric for process uptime after this pr merges for users emitting otlp
otelcol_process_uptime # the metric for process uptime after this pr merges for users scraping via prometheus
process_uptime # the metric for process uptime prior to this merges for users emitting otlp
```

Rather than creating a third potential for this metric name, if the separator were `_` we would simply have this state:

```
otelcol_process_uptime # after this change, no matter if a user is sending OTLP or Prometheus
process_uptime # the metric for process uptime prior to this merges for users emitting otlp
```

Creating a third state disincentivizes users from sending OTLP (because it would break their dashboards/alerts) or upgrading their collector further for the same reason.

We already do not follow the conventions mentioned [here](https://opentelemetry.io/docs/specs/semconv/general/metrics/#instrument-naming), see naming for time and utilization. Prometheus currently rules the metric world for metric naming, independently of the recommendations made by our own semantic convention today. Given that, I think for legacy conversions we should always be opting for not breaking users by following the existing convention. We should reconsider this once semantic convention exists for not only metric naming recommendations (when to use a . or a _), but also for pipeline monitoring.

Reply:
Right the problem is see with using `_`, is that today, changing a collector from emitting metrics using prometheus to OTLP has a problem with metrics we don't control.

As an example, the metric for grpc instrumentation changes from 

```
otelcol_rpc_server_requests_per_rpc
```

to:

```
rpc.server.responses_per_rpc
```

This is because the prometheus metric replaces `.` with `_` (as it does for other collector metrics `/`). Replacing the separator with `_` instead of `.` means users will still have to contend with a broken metric, it's just not clear which metrics will be broken to them, as they may not know what constitutes an instrumentation metric vs a collector metric. I opted to go with `.` as it at least aligns with the opentelemetry semantic convention.

Reply:
> Creating a third state disincentivizes users from sending OTLP (because it would break their dashboards/alerts) or upgrading their collector further for the same reason.

It is true that breaking the name is going to disincentivize users, but I don't think we want to hold ourselves to prometheus naming standards when OTel has its own naming standard.

I can see an argument for waiting until more semantic conventions are stabilized before doing the breaking change to require only 1 breaking change for our end users.

I am ok keeping this change behind the `useOtelWithSDKConfigurationForInternalTelemetry` since it is still alpha.

Reply:
yeah my other concern is that by changing the convention for the collector's own metrics, we actually invalidate any documents for stable/beta components that mention existing metrics. [these documents](https://github.com/search?q=repo:open-telemetry/opentelemetry-collector+otelcol_+language:Markdown&type=code&l=Markdown) mention the important metrics for various components, none mentioning that these are "prometheus convention" but rather that they are the important metrics for monitoring your collector. By creating a dual state, all of these documents would need to be edited as part of this change and then edited again upon guidance from the OTEP. 

Similarly, I would argue that by not keeping the metrics consistent we are introducing a bug to the collector – we would not be able to make the `useOtelWithSDKConfigurationForInternalTelemetry` flag beta because then anyone emitting metrics via OTLP would see incorrect metric names for components like the batch processor.

Reply:
@codeboten is there a world where we handle switch to the OTEL SDK in 2 steps:
1. Get the `useOtelWithSDKConfigurationForInternalTelemetry` to `Stable` without breaking the emitted metric names:
    - this gets us off the prometheus receivers/opencensus dependency and aligned with the project the collector is a part of.
2. Redo the metric names to meet OTel stable metric conventions.

Reply:
Ya let's discuss some more at the SIG